packages/generator-common/src/file-writer/create-file.ts

```diff
@@ -1,9 +1,138 @@
+import { resolve, join, parse } from 'path';
 import { promises } from 'fs';
-import { join } from 'path';
-import { unixEOL, codeBlock, ErrorWithCause } from '@sap-cloud-sdk/util';
+import {
+  codeBlock,
+  createLogger,
+  ErrorWithCause,
+  unixEOL
+} from '@sap-cloud-sdk/util';
+import prettier, {
+  BuiltInParserName,
+  Options as PrettierOptions
+} from 'prettier';
 import { getCopyrightHeader } from '../util';
 
-const { writeFile } = promises;
+const { writeFile, readFile } = promises;
+const logger = createLogger('create-file');
+
+/**
+ * Options for the creation of a file.
+ * @internal
+ */
+export interface CreateFileOptions {
+  /**
+   * Flag to indicate if the file is overwritten.
+   */
+  overwrite?: boolean;
+  /**
+   * Flag to indicate if copy write header is added.
+   */
+  withCopyright?: boolean;
+  /**
+   * Path to the prettier config with respect to the process.cwd().
+   */
+  prettierConfigPath?: string;
+  /**
+   * Flag to indicate if the file is prettified before.
+   */
+  usePrettier?: boolean;
+}
+
+/**
+ * @internal
+ */
+export const defaultPrettierConfig: PrettierOptions = {
+  singleQuote: true,
+  trailingComma: 'none',
+  arrowParens: 'avoid',
+  endOfLine: 'lf'
+};
+
+let prettierConfigCache: PrettierOptions | undefined;
+
+/**
+ * @internal
+ */
+export function clearPrettierConfigCache(): void {
+  prettierConfigCache = undefined;
+}
+
+async function readPrettierConfig(
+  prettierConfigPath: string | undefined
+): Promise<PrettierOptions> {
+  if (prettierConfigCache) {
+    return prettierConfigCache;
+  }
+
+  if (prettierConfigPath) {
+    const resolved = resolve(process.cwd(), prettierConfigPath);
+    try {
+      const config = await readFile(resolved, { encoding: 'utf-8' });
+      prettierConfigCache = JSON.parse(config);
+    } catch (e) {
+      logger.warn(
+        `Prettier config file not found: ${resolved} - default is used.`
+      );
+      prettierConfigCache = defaultPrettierConfig;
+    }
+  }
+
+  return prettierConfigCache || defaultPrettierConfig;
+}
+
+const fileParserMap: Record<string, BuiltInParserName> = {
+  ts: 'typescript',
+  md: 'markdown',
+  json: 'json',
+  js: 'espree',
+  mdx: 'mdx',
+  yml: 'yaml',
+  yaml: 'yaml',
+  'd.ts': 'typescript',
+  'js.map': 'json',
+  'd.ts.map': 'json'
+};
+
+/**
+ * This method considers also double dots like `.map.js`
+ * @param fileName
+ * @returns The complete file extension containing multiple dots
+ * @internal
+ */
+export function getFileExtension(fileName: string): string {
+  const fileType = parse(fileName).base.split('.').slice(1).join('.');
+  return fileType;
```

Comment:
[pp] Maybe return directly?
```suggestion
  return parse(fileName).base.split('.').slice(1).join('.');
```